### DISCUSSION TITLE
Bias field correction

### DISCUSSION THREAD

**Discussion by MartinKjunior** [dipy/dipy#3388] [3. Community Lounge]
I was reading through https://workshop.dipy.org/documentation/1.7.0/examples_built/23_segmentation/tissue_classification/ where the data use has been "skull-stripped and bias field corrected". Skull-stripping can be done using `median_otsu`, but is there any tool in DiPy to do bias field correction? Currently when using DiPy on a diffusion dataset I use `motion_correction`, `gibbs_removal`, `mppca`, `median_otsu` and `TensorModel.fit()`. Are there any other tools in DiPy that I should be using to process the data if I want to fit a DTI model?

**Comment by pjsjongsung** [comment]
The example is talking about a T1 image, which can be skull stripped using `dipy.nn.evac` Unfortunately we do not have a tutorial for that yet but it would be something along this line

```
from dipy.nn.evac import EVACPlus
evac = EVACPlus()
mask = evac.predict(T1, affine, voxsize=voxsize)
```

Also for bias field correction, we have DeepN4
```
from dipy.nn.deepn4 import DeepN4
deepn4 = DeepN4()
deepn4_model.fetch_default_weights()
bias_corrected_T1 = deepn4(T1, affine, voxsize=voxsize)
```

We suggest choosing the pipeline based on your data. For example, running `motion_correction` or `gibbs_removal` would depend on if you actually need them when you see your data.

For denoising, we recommend trying out different denoising methods as well, such as `patch2self` and finding the best one that works for you. The denoising tutorials should be able to serve as some guidelines on what method might be best for you.

`median_otsu` will be required since we do need skull stripped data for any diffusion models.

Overall, I believe you have a good idea of what things you need in the pipeline, and if you run into some things you want to solve in your data, we can coordinate more on what we can add to your pipeline to help solve this.

**Reply by MartinKjunior** [reply]
Thank you for the reply. I was mostly interested in any functions I may have missed when writing my pipeline. I do offer the option to choose different steps, by default I don't do `gibbs_removal` and I only do up to rigid `motion_correction` which seems to work okay on animal data. Unfortunately, I cannot correct for eddy currents with DiPy, but then again FSL doesn't always work for my data so I'm not using it either.

**Reply by MartinKjunior** [reply]
@pjsjongsung I've found the `dipy.nn.cnn_1d_denoising` module, is there any user guide that would show me how to use this on my dataset?

**Reply by skoudoro** [reply]
Just for info, in our lab, we replace eddy currents with motion_correction + patch2self and it does a great job overall

**Reply by MartinKjunior** [reply]
@skoudoro That's good to hear. In my tests on low SNR animal data I ran all available denoising on two brains and found that `mppca` removed less structure than `patch2self`, so I'm doing `mppca` for now and that seems to work alright. `motion_correction` tends to not give great results after the affine step, so I'm just going up to rigid. I've heard about `patch2self` from people working on PreQual and it seems to do well on slightly higher SNR data.

**Reply by pjsjongsung** [reply]
> @pjsjongsung I've found the `dipy.nn.cnn_1d_denoising` module, is there any user guide that would show me how to use this on my dataset?

I don't think there is a trained model for this. The file provides a general framework of how to train a 1d denoising model. For your [reference](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0274396)

**Reply by MartinKjunior** [reply]
> > @pjsjongsung I've found the `dipy.nn.cnn_1d_denoising` module, is there any user guide that would show me how to use this on my dataset?
> 
> I don't think there is a trained model for this. The file provides a general framework of how to train a 1d denoising model. For your [reference](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0274396)

Ah I see, thanks for the response.
